### PR TITLE
doc: Use the term application instead of sample

### PR DESCRIPTION
**Pull request by Avalei**
ref: NCSDK-10594

For the Getting Started sections, use "application" to refer to both
samples and applications for a more unified use of terminology.

Signed-off-by: Deidre Casey <deidre.casey@nordicsemi.no>

### CODE REVIEW

**Review by FrancescoSer — APPROVED** [review]

**Comment by wiba-nordic** [comment]
Should the "For simplicity, this guide will refer to both samples and applications as “applications”." or something similar be in the Testing an application and the Modifying an application sections?

File: doc/nrf/gs_programming.rst [hunk 1]
```diff
@@ -153,15 +154,15 @@ Complete the following steps to build |NCS| projects on the command line after c
 
          The Toolchain Manager dropdown menu options
 
-#.    Go to the specific sample or application directory.
-      For example, to build the :ref:`at_client_sample` sample, run the following command to navigate to the sample directory:
+#.    Go to the specific application directory.
+      For example, to build the :ref:`at_client_sample` sample, run the following command to navigate to the application directory:
```
Comment (peknis) [inline doc/nrf/gs_programming.rst:158]: Not so sure about this as the directory name is "samples".
Maybe something like: ... navigate to its directory.

**Comment by Avalei** [comment]
> 
> 
> Should the "For simplicity, this guide will refer to both samples and applications as “applications”." or something similar be in the Testing an application and the Modifying an application sections? The sections are probably read in consecutive order, but it is possible for someone to jump to the other two as well.

Good call @wiba-nordic , I just added that line to the other two sections.

**Review by wiba-nordic — APPROVED** [review]

File: doc/nrf/gs_programming.rst (hunk 1 above)
Comment (Avalei) [reply doc/nrf/gs_programming.rst]: I was going to do just that, but then second-guessed myself. I agree though, I think "its directory" is better. Adding the change now.

**Review by peknis — APPROVED** [review]

File: doc/nrf/gs_modifying.rst [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
-##############################
+Modifying an application
+########################
 
 .. contents::
    :local:
    :depth: 2
 
-After programming and testing a sample application, you probably want to make some modifications to the application, for example, add your own files with additional functionality, change compilation options, or update the default configuration.
+For simplicity, this guide will refer to both :ref:`samples<samples>` and :ref:`applications<applications>` as "applications".
```
Comment (greg-fer) [inline doc/nrf/gs_modifying.rst:10]: You can put this into shortcuts.txt and use a replacement tag here and on the programming and testing pages.

Comment (Avalei) [reply doc/nrf/gs_modifying.rst]: Updated with your suggestion.

**Comment by wiba-nordic** [comment]
`shortcuts.txt` needs to end with an empty line.

**Review by crfosse — APPROVED** [review]
Looks good to me!

**Pull request merged**